### PR TITLE
Allow users to specify an explicit port for publishing (#97)

### PR DESCRIPTION
**Pull request by jeffrosenberg**
This PR creates a new `UseTargetServerAndPort` method, called only if the user passes in a port. If a port isn't specified, rather than using a default port of 1433, we just fall back to the existing method of not including the port in the connection string at all.

### CODE REVIEW

**Comment by jeffrosenberg** [comment]
@jmezach the CI build is failing on the "Setup .NET Core" step, which this PR doesn't touch at all. Any idea what's going on there? I tried rerunning in case it was just a temporary environment glitch, but it didn't help. I can try rerunning it again in a few hours.

**Comment by jmezach** [comment]
@jeffrosenberg Looks like something has changed on the GitHub Actions side of things. Due to a security vulnerability they have disabled certain features which we are apparently using. I also see that we're still using a custom task to install the .NET SDK on the build agent. We're also still using RC2 of the .NET 5.0 SDK.

Guess we'll have to do some maintenance work here. I'll get on to that when I have some time ;).

**Comment by jmezach** [comment]
So I've modified the workflow to use the default task to install .NET SDK's, which now supports installing multiple versions and that at least gets us past that step in the pipeline. But now it seems to be failing on something that's in the `sqltoolsservice` project that we took a dependency on to fix the SQLCMD file merging. I'll try to update that to a newer version to see if that fixes things.

**Comment by jmezach** [comment]
After updating a whole bunch of stuff everything seems to work now. I still had a failing job, but that was due to a connection reset so I'm guessing that it is a temporary issue.

**Comment by jeffrosenberg** [comment]
@jmezach thanks, looks like everything passes now! Sorry my simple PR led to you having to make 6 commits 🤣

**Comment by jmezach** [comment]
@jeffrosenberg No problem. We'd have to fix this eventually anyway. It is just too bad that there's no way to test out these changes locally ;).

**Review by ErikEJ — APPROVED** [review]

File: src/DacpacTool/Program.cs [hunk 1]
```diff
@@ -47,6 +47,7 @@ static async Task<int> Main(string[] args)
             {
                 new Option<FileInfo>(new string[] { "--input", "-i" }, "Path to the .dacpac package to deploy"),
                 new Option<string>(new string[] { "--targetServerName", "-tsn" }, "Name of the server to deploy the package to"),
+                new Option<string>(new string[] { "--targetPort", "-tprt" }, "Port number to connect on (leave blank for default)"),
```
Comment (jmezach) [inline src/DacpacTool/Program.cs:50]: I think we can have an Option<int> here so that we don't have to do any parsing ourselves.

**Review by jmezach — APPROVED** [review]
LGTM. Only 1 comment.

File: src/DacpacTool/Program.cs (hunk 1 above)
Comment (jeffrosenberg) [reply src/DacpacTool/Program.cs]: I've pushed a change, but C# isn't my primary language, so let me know if it's not the idiomatic way to do things 🙂

File: src/DacpacTool/Program.cs [hunk 2]
```diff
@@ -197,7 +198,14 @@ private static int DeployDacpac(DeployOptions options)
                     }
                 }
 
-                deployer.UseTargetServer(options.TargetServerName);
+                if (options.TargetPort != null)
```
Comment (jmezach) [inline src/DacpacTool/Program.cs:201]: Perhaps more idiomatic C# would be to do `if (options.TargetPort.HasValue)` here ;).

Comment (jeffrosenberg) [reply src/DacpacTool/Program.cs]: Made that change, I believe we should be good to merge this?

**Comment by jmezach** [comment]
Yes I'm okay with merging this. Feels like a new feature so do we want to ship this as 1.10.0?

**Comment by jmezach** [comment]
@jeffrosenberg Thanks for your contribution!

**Comment by jmezach** [comment]
@jeffrosenberg Would you mind adding this to the README as well? I noticed we didn't change that as part of this PR.

**Pull request merged**